src/chunk/mod.rs

```diff
@@ -179,14 +180,18 @@ impl Chunk {
                 //   .await
                 //   .expect("write chunk header");
                 trace!(target: "chunk::write", "header: {:02x?}", &header.buf);
-                writer.write(&header).await.expect("write payload");
+                let header_size = writer.write(&header).await.expect("write header");
+                bytes_written += header_size;
 
                 trace!(target: "chunk::write", "payload: {:02x?}", &buf);
-                writer.write(&buf).await.expect("write payload");
+                let data_size = writer.write(&buf).await.expect("write payload");
+                bytes_written += data_size;
+
+                // TODO: use write_exact to detect socket close
             } // Chunk::Msg
         }; // match chunk
-
-        Ok(_bytes_written)
+        assert_eq!(bytes_written <= (u32::max_value() as usize), true);
```

Comment:
the Chunk "message length" is 24 bits so should really check earlier that data_size < 2^24 
not sure if an assert is the right way to go here... I've only used that in tests, should probably return an error (in a complete implementation this would be a case where the message would be split across chunks)